src/frontend/js/widgets/Dashboard/components/DashboardItem/Enrollment/ProductCertificateFooter/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { FormattedMessage } from 'react-intl';
+import PurchaseButton from 'components/PurchaseButton';
+import { Icon, IconTypeEnum } from 'components/Icon';
+import { CourseLight, Order, Product, ProductType } from 'types/Joanie';
+import { useOrders } from 'hooks/useOrders';
+import { CourseProductProvider } from 'contexts/CourseProductContext';
+import DownloadCertificateButton from 'components/DownloadCertificateButton';
+import { useCertificate } from 'hooks/useCertificates';
+import CertificateStatus from '../../CertificateStatus';
+
+const messages = {
+  buyProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.buyProductCertificateLabel',
+    description: 'Label on the enrollement row that propose to buy a product of type certificate',
+    defaultMessage: 'A exam that provide a certificate can be purchase for this course.',
+  },
+  downloadProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.downloadProductCertificateLabel',
+    description: 'Label on the enrollement row that propose to download a certificate',
+    defaultMessage: 'A certificate is available for download.',
+  },
+  pendingProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.pendingProductCertificateLabel',
+    description: 'Label on the enrollement when a product of type certificate have been bought',
+    defaultMessage: 'Finish this course to obtain your certificate.',
+  },
+};
+
+interface ProductCertificateRowProps {
+  course: CourseLight;
+  product: Product;
+}
+
+const ProductCertificateFooter = ({ course, product }: ProductCertificateRowProps) => {
+  if (product.type !== ProductType.CERTIFICATE) {
+    return null;
+  }
+  const { items: orders } = useOrders({ product: product.id, course: course.code });
+  // FIXME(rlecellier): what to do if we have multiple orders ?
+  const order: Order | null = orders && orders.length ? orders[0] : null;
```

Comment:
A user can only have one validated order per course/product couple

Reply:
ok.
I'll add the state filter on validated and throw a exception if there is more than one.

Reply:
You should retrieve `VALIDATED` and `PENDING` orders in order to prevent user to purchase again a certificate if it has already a order blocked for any reason.